integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/ContentDisposition.java

```diff
@@ -0,0 +1,261 @@
+
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.ISO_8859_1;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.format.DateTimeFormatter.RFC_1123_DATE_TIME;
+
+@Value
+@Builder
+public final class ContentDisposition {
+
+    private final String type;
+    private final String name;
+    private final String filename;
+    private final Charset charset;
+    private final Long size;
+    private final OffsetDateTime creationDate;
+    private final OffsetDateTime modificationDate;
+    private final OffsetDateTime readDate;
+
+
+    /**
+     * Return the header value for this content disposition as defined in RFC 2183.
+     *
+     * @see #parse(String)
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        if (this.type != null) {
+            sb.append(this.type);
+        }
+        if (this.name != null) {
+            sb.append("; name=\"");
+            sb.append(this.name).append('\"');
+        }
+        if (this.filename != null) {
+            if (this.charset == null || StandardCharsets.US_ASCII.equals(this.charset)) {
+                sb.append("; filename=\"");
+                sb.append(this.filename).append('\"');
+            } else {
+                sb.append("; filename*=");
+                sb.append(encodeHeaderFieldParam(this.filename, this.charset));
+            }
+        }
+        if (this.size != null) {
+            sb.append("; size=");
+            sb.append(this.size);
+        }
+        if (this.creationDate != null) {
+            sb.append("; creation-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.creationDate));
+            sb.append('\"');
+        }
+        if (this.modificationDate != null) {
+            sb.append("; modification-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.modificationDate));
+            sb.append('\"');
+        }
+        if (this.readDate != null) {
+            sb.append("; read-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.readDate));
+            sb.append('\"');
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Return an empty content disposition.
+     */
+    public static ContentDisposition empty() {
+        return new ContentDisposition("", null, null, null, null, null, null, null);
+    }
+
+    /**
+     * Parse a {@literal Content-Disposition} header value as defined in RFC 2183.
+     *
+     * @param contentDisposition the {@literal Content-Disposition} header value
+     * @return the parsed content disposition
+     * @see #toString()
+     */
+    public static ContentDisposition parse(String contentDisposition) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "parse" is 14 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)